### ISSUE TITLE
Socket REPL

### ISSUE THREAD

**Issue by thheller** [cursive-ide/cursive#1672]
I'm looking for a way for Cursive to connect to a remote TCP Socket and treat that as a REPL.

It would only need to display everything that comes from the Socket as plain text unmodified. No syntax highlighting or post-processing of any kind is required.

The input should also be sent as-is and does not need to be repeated into the output window.

It would be very convenient for the input editor to support some basic features like syntax highlighting and paredit/parinfer. Beyond that it would not need to provide anything like auto-completions or any other intents.

It would be nice if the REPL actions of "Send X to REPL" or "Load File in REPL" worked as well.

I can currently only kind of fake this behaviour by starting an extra clojure.main REPL and running other utilities there to connect to the other JVM. But Cursive still gets in the way of this by wrapping what is sent into the REPL window, plus I have a bunch of extra JVM running.

I tried the nREPL remote but that tries to `load-file` on startup and always causes an error. The code cannot be loaded as this is a CLJS REPL but I cannot tell Cursive that before connecting and it always resets back to CLJ when starting the connect.

**Comment by devurandom** [comment]
Is it correct that this has been implemented in the meantime? https://cursive-ide.com/userguide/repl.html#remote-repls

**Comment by cursive-ide** [comment]
@devurandom No, since Cursive uses a prepl-like protocol over the socket. What Thomas is asking for is a plain REPL that doesn't do that, i.e. a pure streaming one.
